Provider/src/FirebirdSql.EntityFrameworkCore.Firebird/Migrations/FbMigrationsSqlGenerator.cs

```diff
@@ -15,16 +15,377 @@
+			{
+				builder.Append($"ALTER TABLE {identifier} ALTER COLUMN ");
+				builder.Append(Dependencies.SqlGenerationHelper.DelimitIdentifier(operation.Name));
+
+				if (operation.DefaultValue != null)
+				{
+					var typeMapping = Dependencies.TypeMapper.GetMapping(operation.DefaultValue.GetType());
+					builder.Append(" SET DEFAULT ")
+						.Append(typeMapping.GenerateSqlLiteral(operation.DefaultValue))
+						.AppendLine(Dependencies.SqlGenerationHelper.BatchTerminator);
+				}
+				else if (!string.IsNullOrWhiteSpace(operation.DefaultValueSql))
+				{
+					builder.Append(" SET DEFAULT ")
+						.Append(operation.DefaultValueSql)
+						.AppendLine(Dependencies.SqlGenerationHelper.BatchTerminator);
+				}
+				else
+				{
+					builder.Append(" DROP DEFAULT;");
+				}
+			}
+
+			EndStatement(builder);
+		}
+
+		protected override void Generate(CreateIndexOperation operation, IModel model, MigrationCommandListBuilder builder, bool terminate)
+		{
+			var method = (string)operation[FbAnnotationNames.Prefix];
+
+			builder.Append("CREATE ");
+
+			if (operation.IsUnique)
+			{
+				builder.Append("UNIQUE ");
+			}
+
+			builder
+				.Append("INDEX ")
+				.Append(Dependencies.SqlGenerationHelper.DelimitIdentifier(operation.Name))
+				.Append(" ON ")
+				.Append(Dependencies.SqlGenerationHelper.DelimitIdentifier(operation.Table, operation.Schema));
+
+			if (method != null)
+			{
+				builder
+					.Append(" USING ")
+					.Append(method);
+			}
+
+			builder
+				.Append(" (")
+				.Append(ColumnList(operation.Columns))
+				.Append(")");
+
+			if (terminate)
+			{
+				builder.AppendLine(Dependencies.SqlGenerationHelper.StatementTerminator);
+				EndStatement(builder);
+			}
+		}
+
+		protected override void Generate(
+			CreateIndexOperation operation,
+			IModel model,
+			MigrationCommandListBuilder builder)
+		{
+			Generate(operation, model, builder, true);
+		}
+
+		protected override void Generate(DropIndexOperation operation, IModel model, MigrationCommandListBuilder builder)
+		{
+			builder
+				.Append("ALTER TABLE ")
+				.Append(Dependencies.SqlGenerationHelper.DelimitIdentifier(operation.Table, operation.Schema))
+				.Append(" DROP CONSTRAINT ")
+				.Append(Dependencies.SqlGenerationHelper.DelimitIdentifier(operation.Name))
+				.AppendLine(Dependencies.SqlGenerationHelper.StatementTerminator);
+
+			EndStatement(builder);
+		}
+
+		protected override void Generate(
+			RenameColumnOperation operation,
+			IModel model,
+			MigrationCommandListBuilder builder)
+		{
+			builder.Append("ALTER TABLE ")
+				.Append(Dependencies.SqlGenerationHelper.DelimitIdentifier(operation.Table, operation.Schema))
+				.Append(" ALTER ")
+				.Append(Dependencies.SqlGenerationHelper.DelimitIdentifier(operation.Name))
+				.Append(" TO ")
+				.Append(Dependencies.SqlGenerationHelper.DelimitIdentifier(operation.NewName));
+
+			EndStatement(builder);
+		}
+
+		protected override void ColumnDefinition(
```

Comment:
Formatting.

Reply:
Give me a suggestion / example. Because only then will I do the others the way you need to!

Reply:
https://github.com/cincuranet/FirebirdSql.Data.FirebirdClient/pull/72/files#diff-56b8916bb1616666ceb41c2927e4eaa1R41